network/tls.go

```diff
@@ -345,23 +351,30 @@ func makeVerifier(suite Suite, them *ServerIdentity) (verifier, []byte) {
 		// nil) check that the public key advertsied by the far side is the same
 		// as the public key we expect.
 		if them != nil {
-			if len(cert.URIs) > 0 {
-				// see explanation of the URL scheme above for why we prepend :.
-				cn := fmt.Sprintf(":%v", pubToCN(them.Public))
-				found := false
-				for _, u := range cert.URIs {
-					if u.Scheme == "onet-pubkey" && u.Opaque == cn {
-						found = true
-						break
-					}
-				}
-				if !found {
-					return xerrors.Errorf("No onet-pubkey URIs match the expected public key %v", pubToCN(them.Public))
+			if noURIs {
+				err = cert.VerifyHostname(pubToCN(them.Public))
```

Comment:
This call will not pass, no matter what, if you are testing with Go 1.15. The idea of this PR is to provide two implementations, one like v3.2.4 and one like v3.2.5 and mix and match them. That's cool, but then you can only get it to pass with Go 1.14.

I installed Go 1.14 and then ran the unit test like this:
```
brew install go@1.14
/usr/local/Cellar/go@1.14/1.14.7/bin/go test -run TestTLS -v
```